src/layers/ipv4.rs

```diff
@@ -50,21 +50,58 @@ pub fn register_protocol(proto: u8, creator: LayerCreatorFn) -> Result<(), Error
     Ok(())
 }
 
+// Option Structs
```

Comment:
Not sure if the following struct is needed. I don't see it being used (decoded to) anywhere. Am I missing something?